pkg/util/kubernetes/apiserver/hpa_controller.go

```diff
@@ -308,13 +308,23 @@ func (h *AutoscalersController) addAutoscaler(obj interface{}) {
 // Adding the new obj and dropping the previous one is sufficient.
 // FIXME if the metric name or scope is changed in the HPA manifest we should propagate the change
 // to the Global store here
-func (h *AutoscalersController) updateAutoscaler(_, obj interface{}) {
+func (h *AutoscalersController) updateAutoscaler(old, obj interface{}) {
 	newAutoscaler, ok := obj.(*autoscalingv2.HorizontalPodAutoscaler)
 	if !ok {
 		log.Errorf("Expected an HorizontalPodAutoscaler type, got: %v", obj)
 		return
 	}
-	log.Tracef("Updating autoscaler %s/%s", newAutoscaler.Namespace, newAutoscaler.Name)
+	oldAutoscaler, ok := old.(*autoscalingv2.HorizontalPodAutoscaler)
+	if !ok {
+		log.Errorf("Expected an HorizontalPodAutoscaler type, got: %v", old)
+		return
```

Comment:
I'm wondering what's best between returning here, or enqueue the update just in case. The safe path here might be to still consider the update if `newAutoscaler` is valid. Your change is still a significant improvement since when the old object is valid (i expect it is 99% of the time), the ignore logic works